### ISSUE TITLE
Error parsing App Store JSON data (String but found a number instead.)

### ISSUE THREAD

**Issue by Ufosek** [ArtSabintsev/Siren#246]
- **Are you using the [latest version](https://github.com/ArtSabintsev/Siren/releases) of Siren?**:  Yes (3.8.0)
- **What is your app's Bundle ID?**: I used demo bundle - com.apple.AppStoreConnect
- **When was the latest version of your app published to the App Store?**: -
- **Is your app published in the US App Store? If not, what App Store is it published in?**: yes
- **Does Siren work if you plugin your app's BundleID (and CountryCode, if necessary) into the Example App?**: no (it worked few days ago)

---

Yesterday I noticed that I got such error:

```
[Siren] Error parsing App Store JSON data.
Also, the following system level error was returned: typeMismatch(Swift.String, Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "results", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0), CodingKeys(stringValue: "trackId", intValue: nil)], debugDescription: "Expected to decode String but found a number instead.", underlyingError: nil))
```

Earlier I had similar error but logs were different: https://github.com/ArtSabintsev/Siren/issues/241 and after update to 3.8.0 - it was gone and everything was working. Maybe Apple changed something again in API? It looks like apple changed trackId from Int to String and now to Number.

**Comment by ArtSabintsev** [comment]
Ok thanks for all that info! I’ll look into it tonight!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by dklinzh** [comment]
+1
I also encounter this issue again. See #240 may be one of the solutions for Apple API unexpected changes.

**Comment by ArtSabintsev** [comment]
Reverted it back to an Int. I think Apple is realizing their mistake and reverting it. If it happens again, I'll implement the solution in #240.

This is fixed in v3.9.0 which is now tagged and available via CocoaPods.

**Comment by kailahuy** [comment]
version 3.9.2 still bug. 
 "Expected to decode String but found a number instead."
